kernel/nativelib.ml

```diff
@@ -186,5 +192,10 @@ let call_linker ?(fatal=true) env ~prefix f upds =
   match upds with Some upds -> update_locations upds | _ -> ()
 
 let link_library env ~prefix ~dirname ~basename =
-  let f = dirname / !output_dir / basename in
+  (* We try both [output_dir] and [.coq-native], unfortunately from
+     [Require] we don't know if we are loading a library in the build
+     dir or in the installed layout *)
+  let install_location = dirname / dft_output_dir / basename in
```

Comment:
Does this mean that we could avoid the `if` if `Require` (`rec_intern_library`) had kept the physical path?

In any case, for 8.2.1, this seems enough, since, iiuc, the file to link will be in one of these two locations.

Reply:
> Does this mean that we could avoid the `if` if `Require` (`rec_intern_library`) had kept the physical path?

I was wondering the same thing, indeed.

Reply:
Actually that's not a bad idea at all, tho goes a bit far IMO for a fix targeted for a point-release.

I'd like to push a patch unifying the loadpath machinery and actually making it into a standalone lib, so I've added this observation in regards to the `Require` resolver.